src/main/java/mytown/util/TownBlockKey.java

```diff
@@ -0,0 +1,25 @@
+package mytown.util;
+
+public class TownBlockKey {
+    int dim, x, z;
+    public TownBlockKey(int dim, int x, int z) {
+        this.dim = dim;
+        this.x = x;
+        this.z = z;
+    }
+
+    @Override
+    public int hashCode() {
+        return this.dim ^ this.x ^ this.z;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Since a key will never be retrieved and compared to itself, I believe this situation can't occur, so isn't necessary. Correct me if I'm wrong and missing something.

Reply:
I'm not quite sure how the insertion of a key works, it doesnt really matter tho.

Reply:
It doesnt really matter tho, I'm not quite sure how the insertion in a hashmap works, or what does it compare against.